usl_pipeline/usl_pipeline/readers/elevation.py

```diff
@@ -33,24 +34,66 @@ def read_from_geotiff(
         transform = src.transform
         ll_corner = transform * (0, src.height)
         input_nodata = src.nodata
-        final_nodata_value = (
-            no_data_value if no_data_value is not None else input_nodata
-        )
         elv_header = ElevationHeader(
             col_count=src.width,
             row_count=src.height,
             x_ll_corner=ll_corner[0],
             y_ll_corner=ll_corner[1],
             cell_size=transform[0],
-            nodata_value=final_nodata_value,
+            nodata_value=(no_data_value if no_data_value is not None else input_nodata),
             crs=src.crs,
         )
         elv_data = None
 
         if not header_only:
-            data = src.read(band)
+            elv_data = src.read(band)
             if no_data_value is not None:
-                data[data == input_nodata] = no_data_value
-            elv_data = data
+                elv_data[elv_data == input_nodata] = no_data_value
 
         return Elevation(header=elv_header, data=elv_data)
+
+
+def read_from_esri_ascii(
+    file: typing.TextIO,
+    header_only: bool = False,
+    no_data_value: Optional[float] = None,
+) -> Elevation:
+    """Loading elevation raster data from Esri ASCII file/stream.
+
+    Args:
+        file: Binary stream to load from.
+        header_only: Indicates that only header should be loaded, whereas
+            elevation data should be skipped.
+        no_data_value: Optional value to set in the returned data to indicate
+            absence of data. If not supplied, the no-data value defined in the
+            TIFF file itself will be used.
+
+    Returns:
+        Elevation object.
+    """
+    print(type(file))
+    # read 6 header lines
+    header_map = {}
+    for row_index in range(0, 6):
```

Comment:
it's typical to name variables `_` if you don't use them for anything in Python.